src/clib/xtg/grdcp3d_get_vtk_esg_geometry_data.c

```diff
@@ -0,0 +1,324 @@
+*
+* DESCRIPTION:
+*    Get geometry data that is suitable as input to VTK's vtkExplicitStructuredGrid.
+*    See: https://vtk.org/doc/nightly/html/classvtkExplicitStructuredGrid.html#details
+*
+*    Basically what we have to do is build an unstructured grid representation where
+*    all grid cells are represented as hexahedrons with explicit connectivities.
+*    The connectivity table refers into the accompanying vertex table.
+*
+*    In VTK, cells order increases in I fastest, then J, then K.
+*
+*    This function also tries to remove/weld duplicate vertices, but this is a WIP.
+*
+* ARGUMENTS:
+*    ncol, nrow, nlay     i     Dimensions
+*    coordsv              i     Coordinate vector (xtgeo fmt)
+*    zcornsv              i     ZCORN vector (xtgeo fmt)
+*    vert_arr             o     Will be populated with vertex coordinates (XYZ, XYZ, etc)
+*    conn_arr             o     Will be populated with the cell connectivities
+*
+* RETURNS:
+*    The actual number of XYZ vertices written into vert_arr.
+*    Also updates output arrays vert_arr and conn_arr.
+*
+* TODO/ISSUES/BUGS:
+*    - verify/enhance removal of duplicate nodes
+*    - optimize the way cell corners are accessed
+*    - reduce memory usage in GeoMaker helper class
+*
+* LICENCE:
+*    CF XTGeo's LICENSE
+***************************************************************************************
+*/
+
+#include "libxtg.h"
+#include "libxtg_.h"
+#include "logger.h"
+
+
+// Forward declarations of static helpers
+static struct GeoMaker* gm_create(long cellCountI, long cellCountJ, long cellCountK, double* targetVertexBuffer, long* targetConnBuffer);
+static void gm_destroy(struct GeoMaker* gm);
+static void gm_addVertex(struct GeoMaker* gm, long i, long j, long k, double v[3]);
+static long gm_vertexCount(const struct GeoMaker* gm);
+static long gm_connectivityCount(const struct GeoMaker* gm);
+
+static double getTimeStamp_s();
+
+
+long grdcp3d_get_vtk_esg_geometry_data(long ncol,
+                                       long nrow,
+                                       long nlay,
+
+                                       double *coordsv,
+                                       long ncoordin,
+                                       float *zcornsv,
+                                       long nlaycornin,
+
+                                       double *vert_arr,
+                                       long n_vert_arr,
+                                       long *conn_arr,
+                                       long n_conn_arr)
+{
+    //printf("STARTING native code ----------------------------------\n");
+    const double start_s = getTimeStamp_s();
+
+    //printf("ncol=%ld\n", ncol);
+    //printf("nrow=%ld\n", nrow);
+    //printf("nlay=%ld\n", nlay);
+    //printf("ncoordin=%ld\n", ncoordin);
+    //printf("nlaycornin=%ld\n", nlaycornin);
+    //printf("n_vert_arr=%ld  (=> numVertices=%ld)\n", n_vert_arr, n_vert_arr/3);
+    //printf("n_conn_arr=%ld\n", n_conn_arr);
+
```

Comment:
Remove print* if not needed, even if commented out

Reply:
Diagnostic output lines are dow deleted.